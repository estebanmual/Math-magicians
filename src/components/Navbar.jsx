import React from 'react';
import { NavLink } from 'react-router-dom';
import style from '../styling/Navbar.module.css';

function Navbar() {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: 3,
      path: '/quote',
      text: 'Quote',
    },
  ];

  return (
    <nav className={style.navbar}>
      <h1 className={style.navbar__title}>Math Magicians</h1>
      <ul className={style.navbar__list}>
        {links.map((link) => (
          <li key={link.id} className={style.navbar__listEl}>
            <NavLink to={link.path}>{link.text}</NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
}

export default Navbar;
