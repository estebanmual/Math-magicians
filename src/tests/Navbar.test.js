import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<Router><Navbar /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  render(<Router><Navbar /></Router>);
  expect(screen.getByText('Math Magicians')).toBeInTheDocument();
});
