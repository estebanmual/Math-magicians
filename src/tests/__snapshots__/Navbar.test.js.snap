// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<nav
  className="navbar"
>
  <h1
    className="navbar__title"
  >
    Math Magicians
  </h1>
  <ul
    className="navbar__list"
  >
    <li
      className="navbar__listEl"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="navbar__listEl"
    >
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="navbar__listEl"
    >
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
